### PR TITLE
Async db

### PR DESCRIPTION
**Pull request by jpmonettas**
Migrated server from sqlite to postgres db which pushed a almost everything to async rewrite

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -22,12 +22,14 @@
                  [expound "0.8.4"]
                  [funcool/cuerdas "2.2.0"]
                  [garden "1.3.10"]
+                 [honeysql "1.0.444"]
                  [medley "1.3.0"]
-                 [mount "0.1.16"]
+                 [mount "0.1.17-SNAPSHOT"]
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply project.clj:27]: +1 for district0x/mount, I can help you with the release.

File: project.clj [hunk 2]
```diff
@@ -36,7 +38,7 @@
                  [day8.re-frame/forward-events-fx "0.0.6"]
 
                  ;; District General Libraries
-                 [district0x/async-helpers "0.1.3"]
+                 [district0x/async-helpers "0.1.3-SNAPSHOT"]
```
Comment (fbielejec) [inline project.clj:41]: can we avoid SNAPSHOT versions in production releases? They can lead to hard-to-find bugs and should be reserved for local development. Most of our repositories are configured to release non-snapshots.

File: project.clj [hunk 3]
```diff
@@ -117,7 +119,8 @@
          [graphql-tools "4.0.5"]
          [graphql "14.2.1"]
 
-         [better-sqlite3 "5.4.0"]
+         [better-sqlite3 "5.4.0"] ;; TODO: remove this
```
Comment (fbielejec) [inline project.clj:122]: remove?

File: project.clj [hunk 4]
```diff
@@ -191,8 +194,8 @@
                 :output-dir "target/node/out-dev-server"
                 :target :nodejs
                 :optimizations :none
-                :source-map true
-                :source-map-timestamp true
+                ;; :source-map true
```
Comment (fbielejec) [inline project.clj:197]: uncomment?

File: src/district/server/async_db.cljs [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+(defn- map-keys [f m]
+  (into {} (map (fn [[k v]] [(f k) v]) m)))
+
+(defn get-connection
+  "Returns a db connection from the pool."
+  []
+  (.connect (:connection-pool @db)))
+
+(defn release-connection
+  "Returns a db connection from the pool."
```
Comment (fbielejec) [inline src/district/server/async_db.cljs:46]:
```suggestion
  "Returns a db connection to the pool."
```

File: src/district/server/async_db.clj [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+               (cljs.core.async/take! (district.shared.async-helpers/safe-go ~@body)
+                                      (fn [v-or-err#]
+
+                                        (if (cljs.core/instance? js/Error v-or-err#)
+                                          (do
+                                            (district.server.async-db/release-connection ~conn)
+                                            (reject# v-or-err#))
+
+                                          (do
+                                            (district.server.async-db/commit-tx ~conn)
```
Comment (fbielejec) [inline src/district/server/async_db.clj:18]: Shouldn't it release a connection after commiting the tx?

File: project.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpmonettas) [reply project.clj]: done

File: src/ethlance/server/db.cljs [hunk 7]
```diff
@@ -6,14 +6,19 @@
    [com.rpl.specter :as $ :include-macros true]
    [cuerdas.core :as str]
    [district.server.config :refer [config]]
-   [district.server.db :as db]
+   #_[district.server.db :as db]
```
[the remark this answers is not in the document]
Comment (jpmonettas) [reply src/ethlance/server/db.cljs:9]: done

File: project.clj (hunk 2 above)
Comment (jpmonettas) [reply project.clj]: done

File: project.clj (hunk 3 above)
Comment (jpmonettas) [reply project.clj:126]: done

File: project.clj (hunk 4 above)
Comment (jpmonettas) [reply project.clj:201]: done

File: src/district/server/async_db.cljs (hunk 5 above)
Comment (jpmonettas) [reply src/district/server/async_db.cljs:46]: done

File: src/district/server/async_db.clj (hunk 6 above)
Comment (jpmonettas) [reply src/district/server/async_db.clj:18]: Nice catch, fixed  also the rollback case

**Pull request merged**